server/src/test/java/org/opensearch/index/engine/NRTReplicationEngineTests.java

```diff
@@ -315,18 +348,11 @@ public void testCommitSegmentInfos() throws Exception {
         }
     }
 
-    private NRTReplicationEngine buildNrtReplicaEngine(AtomicLong globalCheckpoint, Store store) throws IOException {
+    private NRTReplicationEngine buildNrtReplicaEngine(AtomicLong globalCheckpoint, Store store, IndexSettings settings)
```

Comment:
nit: I see only one engine unit test using remote store. May be we can override this method which uses default settings. This will prevent passing in a settings object for every test.

```suggestion
private NRTReplicationEngine buildNrtReplicaEngine(AtomicLong globalCheckpoint, Store store) throws IOException {
    buildNrtReplicaEngine(globalCheckpoint, store, defaultSettings);
}
private NRTReplicationEngine buildNrtReplicaEngine(AtomicLong globalCheckpoint, Store store, IndexSettings settings) throws IOException {
...
}
```